src/main/java/org/apache/cassandra/sidecar/utils/JsonErrorHandler.java

```diff
@@ -0,0 +1,87 @@
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.ErrorHandler;
+import io.vertx.ext.web.handler.HttpException;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.REQUEST_TIMEOUT;
+
+/**
+ * Handles failures in Sidecar, and provides a detailed JSON response payload for {@link HttpException HttpExceptions}
+ * and {@code REQUEST_TIMEOUT} errors.
+ */
+public class JsonErrorHandler implements ErrorHandler
+{
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handle(RoutingContext ctx)
+    {
+        Throwable t = ctx.failure();
+
+        if (t instanceof HttpException)
+        {
+            handleHttpException(ctx, (HttpException) t);
+        }
+        else if (ctx.statusCode() == REQUEST_TIMEOUT.code())
+        {
+            handleRequestTimeout(ctx);
+        }
+        else
+        {
+            ctx.next(); // let vertx handle other errors
+        }
+    }
+
+    /**
+     * Writes a JSON payload to the response with the {@code Fail} status and the {@code message} provided
+     * by the {@code httpException}. The {@link HttpException#getStatusCode() status code} from the exception
+     * is used for the status code for the response.
+     *
+     * @param ctx           the context for the handling of a request in Vert.x-Web
+     * @param httpException the {@link HttpException} to be handled
+     */
+    private void handleHttpException(RoutingContext ctx, HttpException httpException)
+    {
+        JsonObject payload = new JsonObject()
+                             .put("status", "Fail")
+                             .put("message", httpException.getPayload());
+
+        writeResponse(ctx, httpException.getStatusCode(), payload);
+    }
+
+    /**
+     * Writes a JSON payload to the response with the {@code Request Timeout} status and the
+     * {@link io.netty.handler.codec.http.HttpResponseStatus#REQUEST_TIMEOUT} status code.
+     *
+     * @param ctx the context for the handling of a request in Vert.x-Web
+     */
+    private void handleRequestTimeout(RoutingContext ctx)
+    {
+        JsonObject payload = new JsonObject().put("status", "Request Timeout");
+        writeResponse(ctx, REQUEST_TIMEOUT.code(), payload);
+    }
+
+    /**
+     * Writes the {@code payload} with the given {@code statusCode} to the response.
+     *
+     * @param ctx        the context for the handling of a request in Vert.x-Web
+     * @param statusCode the HTTP status code for the response
+     * @param payload    the JSON payload for the response
+     */
+    private void writeResponse(RoutingContext ctx, int statusCode, JsonObject payload)
+    {
+        HttpServerResponse response = ctx.response();
+        if (!response.ended() && !response.closed())
+        {
+            response.setStatusCode(statusCode)
+                    .putHeader(HttpHeaders.CONTENT_TYPE.toString(), "application/json")
```

Comment:
The content type of the error response is always json. It can conflict with the `Accept` header provided by the client. See https://www.rfc-editor.org/rfc/rfc7231#section-5.3.2
Meanwhile, it looks like the `io.vertx.ext.web.handler.impl.ErrorHandlerImpl` has the logic to encode the response in json and other formats. For example, 

```java
//  register the out-of-box error handler for routes.
    .failureHandler(ErrorHandler.create(vertx))

//  client request with the Accept header
    .putHeader("Accept", "application/json")

//  response body content
//  {"error":{"code":400,"message":"Bad Request"}}
```